bots/psuteam7bot/movement.js

```diff
@@ -0,0 +1,408 @@
+                    if(currentDist < distArr[i])
+                    {
+                        sortedArr.splice(i, 0, {x, y});
+                        distArr.splice(i, 0, currentDist);
+                        break;
+                    }
+                }
+            }
+        }
+    }
+    return sortedArr;
+}
+*/
+
+/*A more simple moveTowards, move to a nearby passable adjacent tile, hopefully closer to destination
+*Input: self        -   The robot unit
+*       destination -   The destination 'position/ location' object {x, y}
+*Output:    retVal  -   A 'position/ location' object {x, y} which is adjacent to self.me 
+*                   -   OR self.me if they are all not passable
+*version of dumberMoveTowards with robot not recording a previous, might get stuck in corner
+*/
+/*
+movement.dumberMoveTowards = (self, destination) => {
+    let fullMap = self.map;
+    let robotMap = self.getVisibleRobotMap();
+    let direction = getRelativeDirection(self.me, destination);
+    let {x, y} = self.me;
+    let candidate = {x : (x+direction.x), y: (y+direction.y)}
+
+
+    let initDirection = initDirection;
+    do{
+        if(isPassable(candidate, fullMap, robotMap))
+            return candidate;
+
+        direction = movement.rotateDirection(direction, 1);
+    }while(direction !== initDirection);
+
+    return self.me;
+}
+*/
+
+/*The most simplest moveTowards, get location of a nearby passable adjacent tile, hopefully closer to destination
+*Input:     location    -   the robot's 'position/ location' object, should be self.me
+*           fullMap     -   the full map, should be self.map or self.getPassableMap()
+*           robotMap    -   robot map, should be self.getVisibleRobotMap()
+*           destination -   The destination 'position/ location' object {x, y}
+*           previous    -   A 'position/ location' object {x, y}, should be self.previous/ the position the robot was in the previous turn
+*Output:    retVal          -   A 'position/ location' object {x, y} which is adjacent to self.me 
+*                           -   OR the passed in location if they are all not passable
+*/
+movement.dumberMoveTowards = (location, fullMap, robotMap, destination, previous) => {
+    let direction = getRelativeDirection(location, destination);
+    let {x, y} = location;
+    let candidate = {x : (x+direction.x), y: (y+direction.y)}
+
+    if(isPassable(candidate, fullMap, robotMap))
+        return candidate;
+
+    let dirA = direction;
+    let dirB = direction;
+    do{
+        dirA = rotateDirection(dirA, 1);
+        dirB = rotateDirection(dirB, -1);
+
+        let candidateA = {x : (x+dirA.x), y: (y+dirA.y)}
+        if(candidateA !== previous && isPassable(candidateA, fullMap, robotMap))
+            return candidateA;
+
+        let candidateB = {x : (x+dirB.x), y: (y+dirB.y)}
+        if(candidateB !== previous && isPassable(candidateB, fullMap, robotMap))
+            return candidateB;
+    }while(dirA !== direction);
```

Comment:
This is looping endlessly for me. Not positive about equalities in Javascript, but may have to do `dirA.x === direction.x && dirA.y === direction.y`.